### ISSUE TITLE
Plotting and Metrics + Mitosis Info

### ISSUE THREAD

**Issue by MalachiteWind** [MalachiteWind/PyDmDeep#6]
@abjohnson117  Here are the things that would be super great to be added on a separate branch! 

**To Do**:
`exp.tensor_data.run()`
- [x] construct plot of singular value weights. _i.e._ highlight how many singular values capture $99$% of the data maybe?

`exp.lstm_train.run()`
- [ ] print/show metric for error between $V$ and $V_{LSTM}$ (you'll have to create $V_{LSTM}$ from the lstm model) from svd of time delay matrix. 
- [ ] Plot original dataset (`dataset["data"]`) and LSTM constructed dataset ($USV^\top_{LSTM}[:,0]$ or something like that). Also the error between the two datasets.

`pydmdeep.models.lstm.LSTMModel`
- [x] can you check if the definitions for `h0` and `c0` make sense in the `forward` call? And make changes as necessary 

# Running Example using Mitosis
To run experiments using [mitosis](https://github.com/Jacob-Stevens-Haas/mitosis/tree/main) (which needs to be pip installed to your python environment)
```
pip install git+https://github.com/Jacob-Stevens-Haas/mitosis
```

Currently to run an experiment with mitosis the general structure looks like:
```bash
mitosis <step_1> <step_2> ... <step_n> \
   -F <save_path> \
   --eval step_1.arg1=<val|str> \
   --eval step_1.arg2=<val|str> \
   ...
   --param step_n.arg1=<keyword from `config`> \
   --param step_n.arg2=<keyword from 'config`> \
   --debug 
```
Things to Note!
- the output of `<step_i>` gets piped as an input into `<step_i+1>`
- for `<save_path>` I usually just call this `trials` and then specifically `trials/<step_i>` if I am running results from step_i. But do what makes sense for you!
- `--debug` is optional. Only needed if you repo is 'dirty', _i.e_ haven't committed or stashed changes in your currenlty in your repo. I would just include it.
- `--eval` (or `-e`) is used for directly trying values, such as `int`, `float`, or short `str`.  _i.e._ `-e step_1.seed=1234` or `-e step_1.lags=10`

- `--param` (or `-p`) is used for slightly more complex, commonly used, (or just annoying to type) values that can be more easily identified by some keyword. As an example for selecting an torch optimizer we might do `-p step_2.opt=SGD` where `SGD` is simply a keyword for the argument `opt` in a `config.py` file containing a dictionary of lookup words: 
```python
# File: config.py
step_2_lookup_dict = {
   "opt": { "SGD": torch.optim.SGD,  "ADAM": torch.optim.ADAM, ...}
}
```

## MWE for our current repo
This currently creates a toy dataset and trains a LSTM network on the right singular vectors of the time delay matrix.
```bash
mitosis data lstm -F trials/lstm --debug \
    -e data.seed=1234 \
    -e data.lags=30 \
    -e data.train_len=0.8 \
    -p lstm.opt=SGD \
    -p lstm.loss=MSE \
    -p lstm.dataloader_kws=small_batch \
    -p lstm.num_layers=small \
    -e lstm.num_epochs=10 \
    -e lstm.seed=1234
```

Where the `data` step corresponds to `exp.tensor_data.run` and `lstm` refers to `exp.lstm_train.run`.  Params (`-p`) keyword dictionaries are stored in `exp.config`.  The way you specify these steps and config files is done in the `pyproject.toml` file.

## Adding your own Mitosis step
This isn't explcitly necessary right now, but just to give you a rounded guide with mitosis the way you specify a _mitosis step_  is in the `pyproject.toml`. Under `[tool.mitosis.steps]` you declare a list of the form `step_i = ["<script_location>:<file_to_run>", "<config_location>:<dict_name>" ]`.  So as an example for this repo we have 

```toml
# File: pyproject.toml
[tool.mitosis.steps]
data = ["exp.tensor_data:run", "exp.config:tensor_data_lookup"]
lstm = ["exp.lstm_train:run", "exp.config:lstm_train_lookup"]
```
